metadata/fetcher/fetcher_test.go

```diff
@@ -0,0 +1,59 @@
+// Copyright 2023 VMware, Inc.
+//
+// This product is licensed to you under the BSD-2 license (the "License").
+// You may not use this product except in compliance with the BSD-2 License.
+// This product may include a number of subcomponents with separate copyright
+// notices and license terms. Your use of these subcomponents is subject to
+// the terms and conditions of the subcomponent's license, as noted in the
+// LICENSE file.
+//
+// SPDX-License-Identifier: BSD-2-Clause
+
+package fetcher
```

Comment:
I see that the coverage is around 70%, can we improve it? If not, perhaps there's a dead code somewhere that should be refactored.

Reply:
There are some if statements that do request and headers checks, which we cannot test out of the `fetcher.DownloadFile` method.

The other part without coverage are the status codes. I added assertion for 200 and this bumped up to 84,6%. I couldn't find a good url for 403 to refer to long term. It could bump up to 90%, so if you have something in the sleeve, I'll add it.